src/pipeline/events.rs

```diff
@@ -52,20 +53,21 @@ pub struct ContactForceEvent {
 // issue).
 /// A set of queues collecting events emitted by the physics engine.
 pub(crate) struct EventQueue<'a> {
-    // Used ot retrieve the entity of colliders that have been removed from the simulation
+    pub world_id: WorldId,
+
+    // Used to retrieve the entity of colliders that have been removed from the simulation
     // since the last physics step.
     pub deleted_colliders: &'a HashMap<ColliderHandle, Entity>,
-    pub collision_events: RwLock<EventWriter<'a, CollisionEvent>>,
-    pub contact_force_events: RwLock<EventWriter<'a, ContactForceEvent>>,
+    pub collision_events: &'a mut RwLock<Vec<CollisionEvent>>,
+    pub contact_force_events: &'a mut RwLock<Vec<ContactForceEvent>>,
 }
 
 impl<'a> EventQueue<'a> {
-    fn collider2entity(&self, colliders: &ColliderSet, handle: ColliderHandle) -> Entity {
+    fn collider2entity(&self, colliders: &ColliderSet, handle: ColliderHandle) -> Option<Entity> {
         colliders
             .get(handle)
             .map(|co| Entity::from_bits(co.user_data as u64))
             .or_else(|| self.deleted_colliders.get(&handle).copied())
-            .expect("Internal error: entity not found for collision event.")
```

Comment:
I imagine that's a good change to allow users to come back from errors, but if we're returning `None`, isn't that a sign we did a system ordering mistake ?